engine/gui/src/gui_script.cpp

```diff
@@ -597,6 +597,171 @@ namespace dmGui
         return 0;
     }
 
+    /*# gets the named property of a specified gui node 
+     * 
+     * Instead of using specific getters such as gui.get_position or gui.get_scale,
+     * you can use gui.get instead and supply the property as a string or a hash.
+     * While this function is similar to go.get, there are a few more restrictions
+     * when operating in the gui namespace. Most notably, only these propertie identifiers are supported:
+     *
+     * - `"position"`
+     * - `"rotation"`
+     * - `"scale"`
+     * - `"color"`
+     * - `"outline"`
+     * - `"shadow"`
+     * - `"size"`
+     * - `"fill_angle"` (pie)
+     * - `"inner_radius"` (pie)
+     * - `"slice9"` (slice9)
+     *
+     * The value returned will either be a vmath.vector4 or a single number, i.e getting the "position"
+     * property will return a vec4 while getting the "position.x" property will return a single value.
+     *
+     * @name gui.get
+     * @param node [type:node] node to get the property for
+     * @param property [type:string|hash|constant] the property to retrieve 
+     * @examples
+     *
+     * Get properties on existing nodes:
+     *
+     * ```lua
+     * local node = gui.get_node("my_box_node")
+     * local node_position = gui.get(node, "position")
+     * ```
+     */
+    int LuaGet(lua_State* L)
```

Comment:
Note about rotation: 
For go.get/set we support quaternions for e.g rotation. But for gui when using gui.set_rotation and gui.get_rotation we still use euler angles, so we should probably still support that here as well?

Reply:
Perhaps we could expose `euler_rotation` as a separate property and have `rotation` operate on quaternions like we do in other places?

Reply:
> Perhaps we could expose `euler_rotation` as a separate property and have `rotation` operate on quaternions like we do in other places?

If we do so, it's a breaking change then

Currently expected beh. for devs in `gui` is `euler` (because this is what `gui.get_rotation()` and `gui.set_rotation()` do). I think for now it's better to keep it this way.

Reply:
I'm not suggesting we change the behaviour of `gui.get_rotation()` and `gui.set_rotation()`. I'm simply suggesting that we make the new `gui.get(node, "rotation")` and `gui.set(node, "rotation", value)` return and take a quaternion value respectively.

So it would be inconsistent with the existing `gui.get_rotation()` and `gui.set_rotation()` functions, but not a breaking change.

Reply:
it feels strange

Reply:
if we do so, we have to introduce `euler` in this PR